tests/t310-basis.c

```diff
@@ -0,0 +1,25 @@
+/// @file
+/// Test creation and distruction of a 2D Simplex non-tensor H1 basis
+/// \test Test creation and distruction of a 2D Simplex non-tensor H1 basis
+#include <ceed.h>
+#include "t310-basis.h"
+
+int main(int argc, char **argv) {
+  Ceed ceed;
+  const CeedInt P = 6, Q = 4, dim = 2;
+  CeedBasis b;
+  CeedScalar qref[dim*Q], qweight[Q];
+  CeedScalar interp[P*Q], grad[dim*P*Q];
+
+  buildmats((CeedScalar *) &qref, (CeedScalar *) &qweight,
```

Comment:
You can ditch the casts. `buildmats(qref, qweight, interp, grad)`.

Reply:
without the casts I get 'warning: passing argument ... from incompatible pointer type'

Reply:
Full code/error message? We should not have casts here.

Reply:
I can ditch the casts and avoid the warning with MALLOC if that is preferable

Reply:
```
/home/jeremy/Dev/libceed/tests/t310-basis.c:14:13: warning: passing argument 1 of ‘buildmats’ from incompatible pointer type [-Wincompatible-pointer-types]
   buildmats(&qref, &qweight, &interp, &grad);
             ^~~~~
In file included from /home/jeremy/Dev/libceed/tests/t310-basis.c:5:
/home/jeremy/Dev/libceed/tests/t310-basis.h:1:34: note: expected ‘CeedScalar *’ {aka ‘double *’} but argument is of type ‘CeedScalar (*)[(0, 8)]’
 staic void buildmats(CeedScalar *qref, CeedScalar *qweight, CeedScalar *interp,
```

Reply:
Never mind; I got it